### PR TITLE
implement route to upload reviews in backend

### LINKED ISSUE #94: Implement a backend proxy to reviews receiver
## Description
We want to provide a proxy to reviews receiver through the backend so that we can use the frontend UI to upload reviews as well.

Frontend task: #93 

## DoD
- [x] The backend `reviews` route has a `POST` endpoint to upload a list of reviews (of `IRawReview` [interface](https://github.com/RevisarAI/RevisarAI/blob/dev/shared-types/types.ts#L40)).
- [x] The route forwards the request, along with the jwt auth token to the `reviews-receiver`
- [x] The `reviews-receiver` upload route can handle jwt auth token (in addition to current api-key) for authentication and uploading reviews (This requires authorizing the user just like in [Auth Middleware](https://github.com/RevisarAI/RevisarAI/blob/dev/backend/src/common/auth.middleware.ts#L23) - This code can be moved to `revisar-backend-utils` for less duplicate code).

### PR DESCRIPTION
**Pull request by Shaked-abdu**
closes #94

### CODE REVIEW

File: backend/src/controllers/review.controller.ts [hunk 1]
```diff
@@ -122,6 +126,19 @@ The customer may also provide a list of previous replies that did not satisfy hi
     }
   }
 
+  async uploadViaReviewsReceiver(req: AuthRequest<{}, IReview>, res: Response) {
```
Comment (liadyacobi) [inline backend/src/controllers/review.controller.ts]: The [swagger you created for the path](#diff-c4e2f08d7622c5006dfa082af66bc53ef599eec8167889a441c68a91189ad78fR318) specifies `req.body` is an array of IReview (`IReview[]`)
This function specifies just `IReview`

File: reviews-receiver/src/routes/batch.router.ts [hunk 2]
```diff
@@ -60,4 +81,36 @@ const batchRouter = Router();
  */
 batchRouter.post('/', checkApiKey, schemaValidationMiddleware({ body: IBatchReviewList }), BatchController.post);
 
+/**
+ * @swagger
```
Comment (liadyacobi) [inline reviews-receiver/src/routes/batch.router.ts:87]: why is this a different route? we can use the batch route that we already have

File: revisar-server-utils/config.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import { z } from 'zod';
+import env from 'dotenv';
+
```
Comment (liadyacobi) [inline revisar-server-utils/config.ts]: does it work?
I think that it requires that every micro service provide this env variable.
Maybe you can make the auth middleware function receive a JWT token.
I want to consult @chenyair

**Review by liadyacobi — CHANGES_REQUESTED** [review]

**Review by liadyacobi — CHANGES_REQUESTED** [review]
Looks great, just wondering about the config env in the npm package.
I think it's not a good practice and we'll have to think of a different implementation to that

File: revisar-server-utils/config.ts (hunk 3 above)
Comment (Shaked-abdu) [reply revisar-server-utils/config.ts]: "Maybe you can make the auth middleware function receive a JWT token." - I did it at first.
But then i understood that in this implementation the config load the envs from the specific microservice that uses the shared util.
So if it used from reviews receiver it load his ACCESS_TOKEN_SECRET, and when it used from the backend it load the ACCESS_TOKEN_SECRET of the backend

File: reviews-receiver/src/routes/batch.router.ts (hunk 2 above)
Comment (Shaked-abdu) [reply reviews-receiver/src/routes/batch.router.ts:87]: I think that the separation of the routes make it easier to use:
batch route include: x-api-key header, businessId in the body
batch user interface include: JWT Token, without businessId in the body (extract from JWT verify func)

**Comment by Shaked-abdu** [comment]
> Looks great, just wondering about the config env in the npm package. I think it's not a good practice and we'll have to think of a different implementation to that

In this implementation the config only loads the env from the microservices envs, for example while you run backend locally you should only specify ACCESS_TOKEN_SECRET in the .env of the backend (revisar-server-utils does not have any .env)

**Comment by liadyacobi** [comment]
> > Looks great, just wondering about the config env in the npm package. I think it's not a good practice and we'll have to think of a different implementation to that
> 
> In this implementation the config only loads the env from the microservices envs, for example while you run backend locally you should only specify ACCESS_TOKEN_SECRET in the .env of the backend (revisar-server-utils does not have any .env)

This is correct, But my concern is for services which might use other middlewares but dont use the AuthMiddleware.
Even if they dont need a JWT secret, they won't be able to use any middleware because it requires this environment variable

File: revisar-server-utils/middlewares.ts [hunk 4]
```diff
@@ -30,3 +41,32 @@ export const schemaValidationMiddleware =
...
+    }
+  }
+}
+
+export const authMiddleware = (req: AuthRequest, res: Response, next: NextFunction) => {
+  const token = extractBearerToken(req);
+  if (!token) return res.status(httpStatus.UNAUTHORIZED).send('No token provided');
+
+  try {
+    const user = <IUserDetails>jwt.verify(token, config.accessTokenSecret);
```
Comment (Shaked-abdu) [inline revisar-server-utils/middlewares.ts]: @liadyacobi We can change to this and remove the config.ts from revisar-server-utils
```suggestion
   const user = <IUserDetails>jwt.verify(token, process.env.accessTokenSecret);
```

Comment (liadyacobi) [reply revisar-server-utils/middlewares.ts]: I still don't think it's a good practice to require env variables in a package.
What we can do is create a function that receives a JWT SECRET and returns a middleware that uses this JWT SECRET to perform authentication.
This way we keep the logic in the package but the env variables in the services.
Something like:
```typescript
export const createAuthMiddleware = (jwtAccessToken: string) =>
   // ... Auth middleware code 
  (req: AuthRequest, res: Response, next: NextFunction) => {
    const token = extractBearerToken(req);
  };
```

File: reviews-receiver/src/routes/batch.router.ts (hunk 2 above)
Comment (liadyacobi) [reply reviews-receiver/src/routes/batch.router.ts:87]: sure sounds find

**Review by liadyacobi — APPROVED** [review]

**Review by liadyacobi — APPROVED** [review]

**Pull request merged**